packages/payment-solana-exact/src/verify.ts

```
@@ -0,0 +1,156 @@
+import {
+  isValidationError,
+  type x402PaymentRequirements,
+} from "@faremeter/types";
+import {
+  parseSetComputeUnitLimitInstruction,
+  parseSetComputeUnitPriceInstruction,
+} from "@solana-program/compute-budget";
+import {
+  findAssociatedTokenPda,
+  parseCreateAssociatedTokenInstruction,
+  parseTransferCheckedInstruction,
+  TOKEN_PROGRAM_ADDRESS,
+} from "@solana-program/token";
+import {
+  address,
+  type CompilableTransactionMessage,
+  type Instruction,
+} from "@solana/kit";
+import { PaymentRequirementsExtra } from "./facilitator";
+
+function verifyComputeUnitLimitInstruction(instruction: Instruction): boolean {
+  if (!instruction.data) {
+    return false;
+  }
+
+  try {
+    parseSetComputeUnitLimitInstruction({
+      programAddress: instruction.programAddress,
+      data: new Uint8Array(instruction.data),
+    });
+    return true;
+  } catch {
+    return false;
+  }
+}
+
+function verifyComputeUnitPriceInstruction(instruction: Instruction): boolean {
+  if (!instruction.data) {
+    return false;
+  }
+
+  try {
+    parseSetComputeUnitPriceInstruction({
+      programAddress: instruction.programAddress,
+      data: new Uint8Array(instruction.data),
+    });
+    return true;
+  } catch {
+    return false;
+  }
+}
+
+async function verifyTransferInstruction(
+  instruction: Instruction,
+  paymentRequirements: x402PaymentRequirements,
+  destination: string,
+): Promise<boolean> {
+  try {
```

For this type of thing, let's keep the try just around the critical section we're worried about causing an exception (i.e. parseTransferCheckdInstruction).  I don't want to swallow errors that are maybe related to validation (which should have been caught earlier).